feat(api): add plugin allowing server to update cookies

ojeytonwilliams opened the pull request:
This plugin allows the server to give new attributes to existing cookies

Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

<!-- Feel free to add any additional description of changes below this line -->

ojeytonwilliams committed:
feat: add plugin for updating cookies

This plugin allows the server to give new attributes to existing cookies

ojeytonwilliams committed:
fix(test): mock env to show defaults are used

ojeytonwilliams committed:
refactor(test): use res.cookies as stricter test

naomi-lgbt commented on api/src/plugins/cookie-update.ts:
```diff
@@ -0,0 +1,37 @@
+      if (!oldCookie) continue;
+
+      const unsigned = unsign(oldCookie);
+      const raw = unsigned.valid ? (unsigned.value as string) : oldCookie;
```
Can we not type the unsign function better?

Something like `{ valid: false, value: null } | { valid: true, value: string }`?

ojeytonwilliams replied:
Yeah, we could do that, but I went down the rabbit hole a little and created a PR to fastify cookie: https://github.com/fastify/fastify-cookie/pull/292

If they're happy with that, we'll get an eslint error when we update and can just drop `as string` at that point.

ojeytonwilliams committed:
refactor: tidy up

Use the decorator, since that's the standard fastify way and should be a
little faster.

Also, mark payload as unused.

ojeytonwilliams committed:
fix: remove unnecessary type assertions

ojeytonwilliams commented:
Just to make sure there were no false positive tests, I rebased this on top of https://github.com/freeCodeCamp/freeCodeCamp/pull/55425 

If that goes in first, I'll clean this up.

naomi-lgbt reviewed (approved):
So much cleaner!
